libConCoCt.py

```diff
@@ -19,29 +37,85 @@
 class Report(object):
     def __init__(self):
         self.parts = []
-    
+
     def add_part(self, report_part):
         self.parts.append(report_part)
-    
+
     def __str__(self):
         ret = ""
         for p in self.parts:
             ret += str(p)
         return ret
 
+    def to_json(self):
```

Comment:
How about extending the [JSONEncoder](https://docs.python.org/3/library/json.html#json.JSONEncoder) class, instead of defining a non standard function?!